opm/parser/eclipse/Deck/Deck.hpp

```diff
@@ -118,6 +120,8 @@ namespace Opm {
             void initUnitSystem() const;
 
             std::vector< DeckKeyword > keywordList;
+            std::shared_ptr<MessageContainer> m_messageContainer;
+
```

Comment:
It can and it should. Additionally, if it is just `MessageContainer` it doesn't have to be explicitly created in the constructor.